### DISCUSSION TITLE
Add ability to Hide or Disable any field

### DISCUSSION THREAD

**Discussion by sfarrowGR** [keystonejs/keystone#7620] [Questions] [answered]
As many fields are populated dynamically (eg: date_created, user_created), the one thing that seems to be really lacking in the field API is the ability to set any field to either be hidden on the UI or disabled.

For example, I do like users to see the original created and updated dates, but they should not be able to edit them directly.  

Likewise, in some cases the "author" of a post should be set based on the person creating the post, but can only be changed by an admin/editor etc.  If you set the access control to false on read, it still shows the field, but with no data.  In the case of a relationship, you can even edit the data, but then get an access denied if you have restricted the update or create access controls.

One option:
If a fields access control for update or create is set to false... that field should display, but be disabled.  

Second Option (Better):
add a isVisible and isDisabled UI flag to every field.

Thanks!

**Comment by Achisingh** [comment, marked as answer]
Hi @sfarrowGR, this is possible by configuring the `fieldMode`. Please refer to [our docs](https://keystonejs.com/docs/apis/fields#common-configuration:~:text=use%20this%20option.-,createView.fieldMode,config%20if%20defined.%20See%20the%20Schema%20API%20for%20details.,-graphql).
